app/src/main/java/com/toolslab/noisepercolator/db/DataProvider.kt

```diff
@@ -8,9 +8,12 @@ import androidx.content.edit
 import com.toolslab.noisepercolator.NoisePercolator
 import com.toolslab.noisepercolator.model.Message
 import com.toolslab.noisepercolator.view.messages.MessageConverter
+import io.reactivex.Observable
+import timber.log.Timber
 
 class DataProvider(private val context: Context = NoisePercolator.applicationContext(),
-                   private val messageConverter: MessageConverter = MessageConverter()) {
+                   private val messageConverter: MessageConverter = MessageConverter(),
```

Comment:
I agree. This was the best way I could find of doing dependency injection without an extra library (yet). At this moment I pass mocks in tests so no objects are instantiated twice.
The idea is to add a library in the future, I wanted to avoid making the learning curve to steep when learning Kotlin.